src/Workspaces/Core/MSBuild.BuildHost/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -391,132 +391,75 @@ public void RemoveDocument(string filePath)
             }
         }
 
-        public void AddMetadataReference(MetadataReference reference, AssemblyIdentity identity)
+        public void AddMetadataReference(string metadataReferenceIdentity, MetadataReferenceProperties properties, string? hintPath)
         {
             if (_loadedProject is null)
             {
                 return;
             }
 
-            if (reference is PortableExecutableReference peRef && peRef.FilePath != null)
-            {
-                var metadata = new Dictionary<string, string>();
-                if (!peRef.Properties.Aliases.IsEmpty)
-                {
-                    metadata.Add(MetadataNames.Aliases, string.Join(",", peRef.Properties.Aliases));
-                }
-
-                if (IsInGAC(peRef.FilePath) && identity != null)
-                {
-                    // Since the location of the reference is in GAC, need to use full identity name to find it again.
-                    // This typically happens when you base the reference off of a reflection assembly location.
-                    _loadedProject.AddItem(ItemNames.Reference, identity.GetDisplayName(), metadata);
-                }
-                else if (IsFrameworkReferenceAssembly(peRef.FilePath))
-                {
-                    // just use short name since this will be resolved by msbuild relative to the known framework reference assemblies.
-                    var fileName = identity != null ? identity.Name : Path.GetFileNameWithoutExtension(peRef.FilePath);
-                    _loadedProject.AddItem(ItemNames.Reference, fileName, metadata);
-                }
-                else // other location -- need hint to find correct assembly
-                {
-                    var relativePath = PathUtilities.GetRelativePath(_loadedProject.DirectoryPath, peRef.FilePath);
-                    var fileName = Path.GetFileNameWithoutExtension(peRef.FilePath);
-                    metadata.Add(MetadataNames.HintPath, relativePath);
-                    _loadedProject.AddItem(ItemNames.Reference, fileName, metadata);
-                }
-            }
-        }
+            var metadata = new Dictionary<string, string>();
+            if (!properties.Aliases.IsEmpty)
+                metadata.Add(MetadataNames.Aliases, string.Join(",", properties.Aliases));
```

Comment:
why collapse the aliases into a comma separated list?  can't we still operate on them all as individual strings without collapsing/exploding strings?  this makes me wary.

Reply:
This is updating the project file and that's the syntax used there, for example:

https://github.com/dotnet/roslyn/blob/4203ffd588ff7a0e05117a5c189649fd3dcc3056/src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj#L27